### PR TITLE
Move javascript html

### PR DESCRIPTION
**Pull request by berggren**
## Move over Javascript and HTML
Move over the Javascript controllers/directives and HTML templates to the refactor branch in order to get them merged.
#### Note
* Moving over in it's current state with the goal of refactor it in a later PR.
* We will move the API calls to services and add tests in the near future. 

Reviewer: @mbushkov
Related issue: #44

### CODE REVIEW

File: timesketch/ui/js/controllers.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+timesketch.controller('ExploreCtrl', function($scope, $http) {
+        $scope.init = function(sketch_id, view_id, timelines) {
+            $scope.sketch_id = sketch_id;
+            $scope.view_id = view_id;
+            $scope.timelines = timelines;
+            $scope.star = false;
+            $scope.filter = {};
+            $scope.filter.indices = $scope.timelines.split(",");
+            $http.get("/api/v1/sketches/" + $scope.sketch_id + "/views/" + $scope.view_id + "/").success(function(data) {
```
Comment (mbushkov) [inline timesketch/ui/js/controllers.js]: I'd suggest implementing a service to work with Timesketch API. Then you can pas the service as a dependency with controller's arguments (i.e. something like function($scope, $http, timesketchApiService)). Having a dedicated service provides much better documentation and makes the code more testable. I.e. here it would look like:
timesketchService.getView($scope.sketch_id, $scope.view_id)).success(...)

Then in the tests you can just mock getView function and not care about the details on how the URL is built.

File: timesketch/ui/templates/home/home.html [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+limitations under the License.
+-->
+
+{% extends "base.html" %}
+
+{% block sidenav %}
+<div id="navigation">
+  <div class="leftnav-title">
+    <form action="{{ url_for('home_views.home') }}" method="post">
+      {{ form.name }}
```
Comment (mbushkov) [inline timesketch/ui/templates/home/home.html]: Mentioning this just in case. It's really confusing to mix server-side templating with client-side templating. IIRC this is a temporary thing and you're going to migrate all the server-side templates to be client-side.

File: timesketch/ui/js/directives.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      restrict: 'A',
+      scope: {
+        filter: '=',
+        search: '=',
+        meta: '='
+      },
+      link: function (scope, elem, attrs) {
+        scope.$watch("filter", function(value) {
+            var i = scope.filter.indices.indexOf(attrs.index);
+            if (i == -1) {
```
Comment (mbushkov) [inline timesketch/ui/js/directives.js]: There's quite a loot of "manual" DOM manipulation here. Any reason not to do the same via traditional directive template?

Comment (mbushkov) [inline timesketch/ui/js/directives.js]: You can have this directive as a div-container if you use ng-transclude.

File: timesketch/ui/js/directives.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+      link: function (scope, elem, attrs) {
+          var timeline_names = scope.meta.timeline_names;
+          elem.addClass("label");
+          elem.css("color", "#999");
+          elem.append(timeline_names[scope.event._index])
+      }
+    }
+});
+
+directives.directive('indexChooser', function() {
```
Comment (mbushkov) [inline timesketch/ui/js/directives.js]: I'd suggest having an app-specific prefix for all timesketch controllers/services/directives registered in global AngularJs namespace. Something like "ts" (i.e. tsIndexChooser).

File: timesketch/ui/js/directives.js [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+                elem.find("input").prop("checked", true);
+
+            }
+            scope.search()
+        })
+      }
+    }
+});
+
+directives.directive('indexDisabler', function() {
```
Comment (mbushkov) [inline timesketch/ui/js/directives.js]: Same here - why not use a directive template?

File: timesketch/ui/js/directives.js [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+                e.find("input").prop("checked", false);
+            }
+            scope.filter.indices = [];
+            scope.search()
+        })
+      }
+    }
+});
+
+directives.directive('indexEnabler', function() {
```
Comment (mbushkov) [inline timesketch/ui/js/directives.js]: And the same template question here.

File: timesketch/ui/templates/sketch/explore.html [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+    <!-- Result card -->
+    <div class="row">
+      <div class="col-md-12">
+        <div class="card">
+          {% raw %}
+          <div ng-show="meta" style="padding:10px;">
+            {{ meta.es_total_count }} events ({{ meta.es_time/1000 }}s)
+          </div>
+          <div ng-repeat="event in events" ng-controller="EventCtrl">
```
Comment (mbushkov) [inline timesketch/ui/templates/sketch/explore.html]: Does it make sense to make this a directive with its own template?

File: timesketch/ui/js/directives.js (hunk 4 above)
Comment (berggren) [inline timesketch/ui/js/directives.js]: Done.

File: timesketch/ui/js/controllers.js (hunk 1 above)
Comment (berggren) [inline timesketch/ui/js/controllers.js]: Agree, and as discussed this will be done in a near future PR, the great Angular refactor.

File: timesketch/ui/templates/home/home.html (hunk 2 above)
Comment (berggren) [inline timesketch/ui/templates/home/home.html]: Agree, and yes there is a WIP to move over to pure client side templates, but there are some things that needs to be done before this can happen. Moving this to a future PR.

File: timesketch/ui/js/directives.js (hunk 3 above)
Comment (berggren) [inline timesketch/ui/js/directives.js]: Agree. I have added TODOs for this and will file an issue to follow up with a bigger refactor in a future PR.

File: timesketch/ui/js/directives.js (hunk 5 above)
Comment (berggren) [inline timesketch/ui/js/directives.js]: Ack. See earlier comment. Will be done in another PR as a larger refactor.

File: timesketch/ui/js/directives.js (hunk 6 above)
Comment (berggren) [inline timesketch/ui/js/directives.js]: Ack. See earlier comment. Will be done in another PR as a larger refactor.

File: timesketch/ui/templates/sketch/explore.html (hunk 7 above)
Comment (berggren) [inline timesketch/ui/templates/sketch/explore.html]: Absolutely. I have added a TODO to fix this. In this PR I want to move the stuff in and get comments and implement the changes in a future PR.

**Comment by berggren** [comment]
As discussed off list, The current code will be merged and then later refactored. The following issue has been created to track this: https://github.com/google/timesketch/issues/46

**Comment by mbushkov** [comment]
LGTM. Changes look good to me and refactoring after the merge is done sounds like a right way to go.
